### PR TITLE
Support for NSX-T 2.4.0

### PR DESCRIPTION
**Pull request by Dyanngg**
## Change summary
- nsx-t versions prior to 2.4.0 will use _ansible-for-nsxt/v1.0.0_ branch and nsx-t 2.4.0 will use the _ansible-for-nsxt/master_ branch.
- Both manager OVA and deployed controllers will serve _nsx-manager_ and _nsx-controller_ roles.
- _basic_topology.yml_ will be modified in runtime based on nsx-t version

### CODE REVIEW

**Comment by kais66** [comment]
it needs a rebase

File: nsxt_yaml/basic_topology.yml [hunk 1]
```diff
@@ -24,11 +22,13 @@
         netmask: "{{hostvars['localhost'].netmask}}"
         admin_password: "{{hostvars['localhost'].nsx_manager_password}}"
         cli_password: "{{hostvars['localhost'].nsx_manager_password}}"
-        ovftool_path: "{{ovftool_bin_path}}"
         path_to_ova: "{{hostvars['localhost'].nsx_image_webserver}}"
         ova_file: "{{hostvars['localhost'].ova_file_name}}"
+        vcenter: "{{hostvars['localhost'].vcenter_ip}}"
```
Comment (kais66) [inline nsxt_yaml/basic_topology.yml]: is this change necessary?

File: tasks/install-nsx-t/get_mo_ref_id.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def modify_line_if_matched(self, line):
 
     def modify_deploy_size_if_matched(self, line):
         new_line = line
-        if line and 'deployment_size' in line:
+        if line and 'deployment_size' in line and 'nsx_manager' not in line:
```
Comment (kais66) [inline tasks/install-nsx-t/get_mo_ref_id.py]: add a comment for this?

File: tasks/install-nsx-t/task.sh [hunk 3]
```diff
@@ -73,12 +92,17 @@ python get_mo_ref_id.py --host $vcenter_ip_int --user $vcenter_username_int --pa
...
+    modify_ansible_tasks_for_unified_appliance
+fi
+
 # Deploy the ovas if its not up
 echo "Installing ovftool"
 install_ovftool
 
 cp ${PIPELINE_DIR}/tasks/install-nsx-t/turn_off_reservation.py ./
 cp ${PIPELINE_DIR}/tasks/config-nsx-t-extras/*.py ./
+
```
Comment (kais66) [inline tasks/install-nsx-t/task.sh]: not needed

File: nsxt_yaml/basic_topology.yml (hunk 1 above)
Comment (Dyanngg) [reply nsxt_yaml/basic_topology.yml]: It is not. But this is the order of param specifications in nsxt-ansible and nsx-t documentation, so I reordered these just to be consistent

File: tasks/install-nsx-t/get_mo_ref_id.py (hunk 2 above)
Comment (Dyanngg) [reply tasks/install-nsx-t/get_mo_ref_id.py]: Sure
